swarm/metrics/flags.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	"github.com/ethereum/go-ethereum/log"
+	gethmetrics "github.com/ethereum/go-ethereum/metrics"
+	metrics "github.com/ethersphere/go-metrics"
+	influxdb "github.com/ethersphere/go-metrics-influxdb"
+	"gopkg.in/urfave/cli.v1"
+)
+
+var (
+	metricsEndpointFlag = cli.StringFlag{
+		Name:  "metricsendpoint",
+		Usage: "metrics backend endpoint",
+		Value: "http://127.0.0.1:8086",
+	}
+	metricsDatabaseFlag = cli.StringFlag{
+		Name:  "metricsdatabase",
+		Usage: "metrics backend database",
+		Value: "metrics",
+	}
+	metricsUsernameFlag = cli.StringFlag{
+		Name:  "metricsusername",
+		Usage: "metrics backend username",
+		Value: "admin",
+	}
+	metricsPasswordFlag = cli.StringFlag{
+		Name:  "metricspassword",
+		Usage: "metrics backend password",
+		Value: "admin",
+	}
+	metricsHostTagFlag = cli.StringFlag{
+		Name:  "metricshosttag",
+		Usage: "metrics host tag",
```

Comment:
What is a "host tag"? Perhaps this could do with a better description.

Reply:
The `host` tag is a `group by` tag within InfluxDB (other monitoring systems also support tags, but I am not sure if they are actually called tags). This is used so that we can group all nodes and average a measurement over all nodes, but also so that we can select a specific node and inspect its own measurements.

Reply:
Ok thanks, this should be explained in the description of the flag.

Reply:
Fixed.